x-pack/plugins/alerting/server/task_runner/inject_action_params.ts

```diff
@@ -7,27 +7,24 @@
 
 import { i18n } from '@kbn/i18n';
 import { RuleActionParams } from '../types';
+import { RuleUrl } from './execution_handler';
 
 export interface InjectActionParamsOpts {
   actionTypeId: string;
   actionParams: RuleActionParams;
-  ruleUrl?: string;
+  ruleUrl?: RuleUrl;
 }
 
 export function injectActionParams({
   actionTypeId,
   actionParams,
-  ruleUrl = '',
+  ruleUrl = {},
 }: InjectActionParamsOpts) {
   // Inject kibanaFooterLink if action type is email. This is used by the email action type
   // to inject a "View alert in Kibana" with a URL in the email's footer.
   if (actionTypeId === '.email') {
-    let path;
-    try {
-      path = new URL(ruleUrl).pathname;
-    } catch (e) {
-      path = '';
-    }
+    // path should not include basePathname since it is part of kibanaBaseUrl already
```

Comment:
Can we double check this?
It looks like `kibanaBaseUrl` is just the `publicBaseUrl` from the config and it cannot have any path on it.

https://github.com/elastic/kibana/blob/65fd7ad260b421387ea2aa253c8f06a7e166695a/x-pack/plugins/alerting/server/plugin.ts#L228

Reply:
The public base URL should include basePath as described [here](https://www.elastic.co/guide/en/kibana/current/settings.html):

> server.publicBaseUrl
>
> The publicly available URL that end-users access Kibana at. Must include the protocol, hostname, port (if different than the defaults for http and https, 80 and 443 respectively), and the [server.basePath](https://www.elastic.co/guide/en/kibana/current/settings.html#server-basePath) (if configured). This setting cannot end in a slash (/).

Reply:
Thanks for the clarification, all look good.